test/Microsoft.AspNet.Mvc.TagHelpers.Test/LinkTagHelperTest.cs

```diff
@@ -451,8 +440,9 @@ public static TheoryData DoesNotRunWhenARequiredAttributeIsMissing_Data
             helper.Process(context, output);
 
             // Assert
-            Assert.NotNull(output.TagName);
```

Comment:
Because `output.TagName` is no longer set to `null` for all cases. It is set to null only for a special case as mentioned here https://github.com/aspnet/Mvc/pull/2829#discussion_r35039322.